### DISCUSSION TITLE
Multiple Relation Row in a View, expand on a list of relations in a view

### DISCUSSION THREAD

**Discussion by Raphael2b3** [pocketbase/pocketbase#5116] [Q&A]
Hello,

i have following tables

```SQL
user (id)
orders(id, customer, product)
products(id)

```
I want to create a View that Allows me to get a JSON object that looks like that

```js

{
 id: "useridxyz",
 products: ["productid1","productid2"]
 expand:[{
  //expand on products
 }]
}
```

This behaviour is seen when creating multiple relation rows in base tables.

i want this data type in a view.

the closest i came up with was

```SQL
SELECT 
  users.id as id,
  SUM(orders.consumed) AS sum_donwloads,
  SUM(products.price * orders.amount) AS sum_costs,
  JSON_GROUP_ARRAY(products.id) AS products
FROM users 
  JOIN orders ON orders.customer = users.id 
  JOIN products ON orders.product = products.id
```

How ever this doesnt allow me to expand on products

Kind regards

**Comment by ganigeorgiev** [comment]
Column expressions (eg. `JSON_GROUP_ARRAY`) are currently always resolved to `json` since they can be anything and are difficult to parse, aka. there is no option at the moment to resolve the `products` field as multiple `relation`. Support for custom view schema mapping exists in the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?filterQuery=view&pane=issue&itemId=32302192) but remains a low priority for now.

You'll have to send separate request(s) to retrieve the related user products, for example:

```js
const productIds = ["id1", "id2", "..."]

const products = await pb.collection("products").getFullList({
    // filter: 'id="id1" || id="id2" || id="..."'
    filter: productIds.map((id) => `id="${id}"`).join("||")
})
```

Depending on your use case and how do you plan to use the aggregation, another option since you want to expand the related items anyway is to not use a view and instead calculate the items you need on the client-side:

```js
const users = await pb.collection("users").getFullList({
    expand: "orders_via_customer.product"
})
```

The above should return all users with their orders and products under the `expand` property (_make sure that your client request satisfies the View API rule of the orders and products collections_).
